qiskit/ignis/logging/ignis_logging.py

```diff
@@ -0,0 +1,321 @@
+
+    def log_to_file(self, **kargs):
+        """
+        This function logs key:value pairs to a log file.
+        Note: Logger name in the log file is fixed (ignis_logging)
+
+        :param kargs: Keyword parameters to be logged (e.g t1=0.02,
+        qubits=[1,2,4])
+        """
+        if not self._file_logging_enabled or self._file_handler is None:
+            return
+
+        Logger.addHandler(self, self._file_handler)
+        logstr = ""
+        for k, v in kargs.items():
+            logstr += "'{}':'{}' ".format(k, v)
+
+        Logger.log(self, 100, logstr)
+
+        Logger.removeHandler(self, self._file_handler)
+
+    def enable_file_logging(self):
+        """
+        Enables file logging for this logger object (note there is a single
+        object for a given logger name
+        """
+        self._file_logging_enabled = True
+
+    def disable_file_logging(self):
+        """
+        Enables file logging for this logger object (note there is a single
+        object for a given logger name
+        """
+        self._file_logging_enabled = False
+
+
+class IgnisLogging:
+    """
+    Singleton class to configure file logging via IgnisLogger.  Logging to file
+    is enabled only if there is a config file present. Otherwise IgnisLogger
+    will behave as a regular logger.
+
+    Config file is assumed to be <user home>/.qiskit/logging.yaml
+
+    Config file fields:
+    ===================
+    file_logging: {true/false}      - Specifies whether file logging is enabled
+    log_file_path: <path>           - path to the log file. If not specified,
+                                        ignis.log will be used
+    max_size:  <# bytes>            - maximum size limit for a given log file.
+                                        If not specified file size is unlimited
+    max_rotations: <count>          - maximum number of log files to rotate
+                                        (oldest file is deleted in case count
+                                        is reached)
+    """
+
+    # TODO: Should we allow to override file settings programmatically ?
+    # (e.g. enable logging)
+    _instance = None
+    _file_logging_enabled = False
+    _log_file = None
+    _max_bytes = 0
+    _max_rotations = 0
+    _log_label = "ignis_logging"
+    _default_datefmt = '%Y/%m/%d %H:%M:%S'
+
+    # Making the class a Singleton
+    def __new__(cls):
+        if IgnisLogging._instance is None:
+            IgnisLogging._instance = object.__new__(cls)
+            IgnisLogging._initialize()
+
+        return IgnisLogging._instance
+
+    @staticmethod
+    def _load_config_file():
+        """
+        Loads and parses the config file
+        :return: a dictionary containing the the settings
+        """
+        config_file_path = os.path.join(os.path.expanduser('~'),
+                                        ".qiskit", "logging.yaml")
+        config = dict()
+        if os.path.exists(config_file_path):
```

Comment:
Can we get a warning if a config file doesn't exist?

Reply:
Yes, good idea